api/store/mongo/device_store.go

```diff
@@ -308,10 +318,23 @@ func (s *Store) DeviceGetByName(ctx context.Context, name, tenant string) (*mode
 }
 
 func (s *Store) DeviceGetByUID(ctx context.Context, uid models.UID, tenant string) (*models.Device, error) {
-	device := new(models.Device)
+	var device *models.Device
+
+	if err := s.cache.Get(ctx, strings.Join([]string{"device", string(uid)}, "/"), &device); err != nil {
+		fmt.Println(err)
+	}
+
+	if device != nil {
+		return device, nil
+	}
+
 	if err := s.db.Collection("devices").FindOne(ctx, bson.M{"tenant_id": tenant, "uid": uid}).Decode(&device); err != nil {
 		return nil, err
 	}
 
+	if err := s.cache.Set(ctx, strings.Join([]string{"device", string(uid)}, "/"), device, time.Minute); err != nil {
+		fmt.Println(err)
```

Comment:
... and here?

Reply:
WIP
https://github.com/shellhub-io/shellhub/blob/94d3a9b492f4599a8dfd95b4668dcd47ccd87b13/api/store/cache/cache_redis.go#L30